applications/GeoMechanicsApplication/custom_conditions/T_microclimate_flux_condition.cpp

```diff
@@ -0,0 +1,483 @@
+    const auto potential_storage =
+        PreviousStorage + TimeStepSize * (precipitation - potential_evaporation);
+
+    auto actual_precipitation = precipitation;
+    auto actual_evaporation = potential_evaporation;
+
+    // Correct the precipitation and evaporation if the storage is out of bounds
+    if (potential_storage > mMaximalStorage)
+    {
+        actual_precipitation =
+            (mMaximalStorage - PreviousStorage) / TimeStepSize + actual_evaporation;
+    }
+    else if (potential_storage < mMinimalStorage)
+    {
+        actual_evaporation = (PreviousStorage - mMinimalStorage) / TimeStepSize + actual_precipitation;
+    }
+
+    WaterFluxes water_fluxes;
+    water_fluxes.evaporation = actual_evaporation;
+    water_fluxes.precipitation = actual_precipitation;
+    return water_fluxes;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+double TMicroClimateFluxCondition<TDim, TNumNodes>::CalculatePotentialEvaporation(
+    unsigned int NodeIndex, double NetRadiation, double SurfaceHeatStorage) const
+{
+    auto& r_geom = this->GetGeometry();
+
+    // Eq 5.35
+    const auto wind_speed = r_geom[NodeIndex].FastGetSolutionStepValue(WIND_SPEED);
+    const auto atmospheric_resistance = 1.0 / (0.007 + 0.0056 * wind_speed);
+
+    // Eq. 5.12
+    const auto atmospheric_temperature =
+        r_geom[NodeIndex].FastGetSolutionStepValue(AIR_TEMPERATURE);
+    const auto saturated_vapor_pressure =
+        6.11 * std::exp(17.27 * atmospheric_temperature / (atmospheric_temperature + 237.3));
+
+    // Eq 5.13
+    const auto vapor_pressure_increment =
+        4098.0 * saturated_vapor_pressure /
+        (std::pow((atmospheric_temperature + 237.3), 2.0));
+
+    // Eq 5.14
+    const auto humidity = r_geom[NodeIndex].FastGetSolutionStepValue(AIR_HUMIDITY);
+    const auto actual_vapor_pressure = humidity / 100.0 * saturated_vapor_pressure;
+
+    // Eq 5.34
+    auto latent_heat_flux =
+        (vapor_pressure_increment * (NetRadiation + mBuildEnvironmentRadiation - SurfaceHeatStorage) +
+         AirHeatCapacity * AirDensity *
+             (saturated_vapor_pressure - actual_vapor_pressure) / atmospheric_resistance) /
+        (vapor_pressure_increment +
+         PsychometricConstant * (1.0 + SurfaceResistance / atmospheric_resistance));
+    latent_heat_flux = std::max(latent_heat_flux, 0.0);
+
+    // Eq 5.36
+    return latent_heat_flux / (WaterDensity * LatentEvaporationHeat);
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TMicroClimateFluxCondition<TDim, TNumNodes>::CalculateRoughness(const ProcessInfo& rCurrentProcessInfo)
+{
+    const auto time_step_size = rCurrentProcessInfo.GetValue(DELTA_TIME);
+
+    const auto& r_geom = this->GetGeometry();
+    const auto current_air_temperature = r_geom[0].FastGetSolutionStepValue(AIR_TEMPERATURE);
+    const auto current_wind_speed =
+        std::max(1.0e-3, r_geom[0].FastGetSolutionStepValue(WIND_SPEED));
```

Comment:
Is it better to have a named constant for 1.0e-3?

Reply:
Agreed, renamed it